src/variables.jl

```diff
@@ -55,6 +88,59 @@ function add_constrained_variable(model::ModelLike, set::AbstractScalarSet)
     return variable, constraint
 end
 
+"""
+    supports_add_constrained_variables(
+        model::ModelLike,
+        S::Type{<:AbstractScalarSet}
+    )::Bool
+
+Return a `Bool` indicating whether `model` supports constraining a vector of
+variables to belong to a set of type `S` either on creation of the vector of
+variables with [`add_constrained_variables`](@ref) or after the variable is
+created with [`add_constraint`](@ref).
+
+By default, if `S` is `Reals` then this function returns `true` and otherwise,
+it falls back to `supports_add_constrained_variables(model, Reals) &&
+supports_constraint(model, MOI.VectorOfVariables, S)` which is the correct
+definition for most models.
+
+## Example
+
+In the standard conic form, the variables are grouped into several cones
+and the constraints are affine equality constraints.
+If `Reals` is not one of the cones supported by the solvers then it needs
+to implement `supports_add_constrained_variables(::Optimizer, ::Type{Reals}) = false`
+as free variables are not supported.
+The solvers should then implement
+`supports_add_constrained_variables(::Optimizer, ::Type{<:SupportedCones}) = true`
+where `SupportedCones` is the union of all cone types that are supported
+but it should not implement
+`supports_constraint(::Type{VectorOfVariables}, Type{<:SupportedCones})`
+as it should return `false`.
+This prevents the user to constrain the same variable in two different cones.
+When a `VectorOfVariables`-in-`S` is added, the variables of the vector
+have already been created so they already belong to given cones.
+The constraint will therefore be bridged by adding slack variables in `S`
+and equality constraints ensuring that the slack variables are equal to the
+corresponding variables of the given constraint function.
+
+Note that there may also be sets for which
+`!supports_add_constrained_variables(model, S)` and
+`supports_constraint(model, MOI.VectorOfVariables, S)`.
+For instance, suppose a solver supports positive semidefinite variable
+constraints and two types of variables: binary variables and nonnegative
+variables. Then the solver should support adding
+`VectorOfVariables`-in-`PositiveSemidefiniteConeTriangle` constraints but it
+should not support creating variables constrained to belong to the
+`PositiveSemidefiniteConeTriangle` as free variables are not supported.
```

Comment:
Reword mention of "free variables". 
Maybe: "because variables in `PositiveSemidefiniteConeTriangle` are not necessarily binary or non-negative".